lib/ransack/adapters/active_record/context.rb

```diff
@@ -183,13 +189,13 @@ def build_correlated_subquery(association)
           join_root = join_constraints.shift
           join_table = join_root.left
           correlated_key = join_root.right.expr.left
+          if (correlated_key.is_a?(Arel::Nodes::Equality))
+            correlated_key = correlated_key.left
+          end
```

Comment:
Are there cases where this is not an Equality node?

Reply:
Yes